### PR TITLE
aya: Raise the RLIMIT_MEMLOCK warning only if failed to create a map

### PR DESCRIPTION
**Pull request by vadorovsky**
Also, mention that setting the RLIMIT_MEMLOCK to a higher value is an
option.

Signed-off-by: Michal Rostecki <vadorovsky@gmail.com>

### CODE REVIEW

File: aya/src/maps/mod.rs [hunk 1]
```diff
@@ -243,6 +280,15 @@ impl Map {
         let c_name = CString::new(name).map_err(|_| MapError::InvalidName { name: name.into() })?;
 
         let fd = bpf_create_map(&c_name, &self.obj.def).map_err(|(code, io_error)| {
+            let kernel_version = {
+                let (major, minor, patch) = crate::sys::kernel_version().unwrap();
```
Comment (alessandrod) [inline aya/src/maps/mod.rs:284]: please add a `use` directive at the top instead of using the full path

File: aya/src/maps/mod.rs [hunk 2]
```diff
@@ -243,6 +280,15 @@ impl Map {
         let c_name = CString::new(name).map_err(|_| MapError::InvalidName { name: name.into() })?;
 
         let fd = bpf_create_map(&c_name, &self.obj.def).map_err(|(code, io_error)| {
+            let kernel_version = {
```
Comment (alessandrod) [inline aya/src/maps/mod.rs:283]: you should compare tuples here like it's done in other places, eg see https://github.com/aya-rs/aya/blob/b2a6f00212997a997799c88ba9022a69d9a0b582/aya/src/programs/cgroup_sock_addr.rs#L77

File: aya/src/maps/mod.rs [hunk 3]
```diff
@@ -222,6 +225,40 @@ impl AsRawFd for MapFd {
...
+            write!(f, "{} KiB", self.0 / 1024)
+        } else {
+            write!(f, "{} MiB", self.0 / 1024 / 1024)
+        }
+    }
+}
+
+/// Raises a warning about rlimit. Should be used only if creating a map was not
+/// successful.
+fn check_rlimit() {
```
Comment (alessandrod) [inline aya/src/maps/mod.rs:244]: I think this should be called `maybe_warn_rlimit`

**Review by alessandrod — CHANGES_REQUESTED** [review]
Looks good, some minor nits

**Comment by vadorovsky** [comment]
@alessandrod Thanks, should be fixed now

**Pull request merged**